hyperspy/_components/voigt.py

```diff
@@ -16,136 +16,99 @@
 # You should have received a copy of the GNU General Public License
 # along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
 
-import numpy as np
 import math
+import numpy as np
+import sympy
 
-from hyperspy.component import Component
+from hyperspy._components.expression import Expression
+from hyperspy._components.gaussian import _estimate_gaussian_parameters
+from distutils.version import LooseVersion
 
 sqrt2pi = math.sqrt(2 * math.pi)
+sigma2fwhm = 2 * math.sqrt(2 * math.log(2))
 
 
-def voigt(x, FWHM=1, gamma=1, center=0, scale=1):
-    """Voigt lineshape.
+class Voigt(Expression):
 
-    The voigt peak is the convolution of a Lorentz peak with a Gaussian peak.
+    r"""Voigt component.
 
-    The formula used to calculate this is::
+    Symmetric peak shape based on the convolution of a Lorentzian and Normal 
+    (Gaussian) distribution. In this case using an approximate formula by David 
```

Comment:
Because the formula to calculate the Voigt component looks complicated at first read, I would first give the definition of the Voigt function with the convolution (as a reference) and then I would give the formula used for the calculation.

Reply:
good idea. done.